### ISSUE TITLE
System.Xml.XmlException: '.', hexadecimal value 0x00, is an invalid character. Line 15, position 1.

### ISSUE THREAD

**Issue by rkantos** [qwertydog/Reddit-Wallpaper-Changer#10]
When starting RWC I receive the following error. It is a fresh install without `%appdata%/Reddit Wallpaper Changer`. RWC is running in the background and it is possible to "bring to front" via task manager but it still doesn't function correctly... UI elements are all over the place and it doesn't appear in task bar or tray at all.

Maybe something to do with region settings? I tried to change to something else but it didn't help.

```
************** Exception Text **************
System.Xml.XmlException: '.', hexadecimal value 0x00, is an invalid character. Line 15, position 1.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.ParseText(Int32& startPos, Int32& endPos, Int32& outOrChars)
   at System.Xml.XmlTextReaderImpl.ParseText()
   at System.Xml.XmlTextReaderImpl.ParseElementContent()
   at System.Xml.XmlTextReaderImpl.Skip()
   at System.Configuration.XmlUtil.StrictSkipToNextElement(ExceptionAction action)
   at System.Configuration.BaseConfigurationRecord.ScanSectionsRecursive(XmlUtil xmlUtil, String parentConfigKey, Boolean inLocation, String locationSubPath, OverrideModeSetting overrideMode, Boolean skipInChildApps)
   at System.Configuration.BaseConfigurationRecord.ScanSectionsRecursive(XmlUtil xmlUtil, String parentConfigKey, Boolean inLocation, String locationSubPath, OverrideModeSetting overrideMode, Boolean skipInChildApps)
   at System.Configuration.BaseConfigurationRecord.ScanSections(XmlUtil xmlUtil)
   at System.Configuration.BaseConfigurationRecord.InitConfigFromFile()


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4515.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
Reddit Wallpaper Changer
    Assembly Version: 1.0.20.0
    Win32 Version: 1.0.20.0
    CodeBase: file:///C:/Program%20Files%20(x86)/Reddit%20Wallpaper%20Changer/Reddit%20Wallpaper%20Changer.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4550.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4536.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4390.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4566.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4190.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Data.SQLite
    Assembly Version: 1.0.116.0
    Win32 Version: 1.0.116.0
    CodeBase: file:///C:/Program%20Files%20(x86)/Reddit%20Wallpaper%20Changer/System.Data.SQLite.DLL
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4536.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.Transactions
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_64/System.Transactions/v4.0_4.0.0.0__b77a5c561934e089/System.Transactions.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.



```

**Comment by qwertydog** [comment]
Hmm, it sounds like the config file has corrupted somehow (looks like it contains 0x00 which is a `NUL` character).

Close the app and try using a text editor to alter the `Reddit Wallpaper Changer.exe.config` file, delete and retype lines 14-16 manually, if you have a text editor like Notepad++ installed it can show non-printable characters so you can find and delete the `NUL` character.

Otherwise, close the app and delete the `Reddit Wallpaper Changer.exe.config` file, then restart the app and see if it gets regenerated correctly.

If that doesn't work, try re-downloading the installer from Github again, uninstall the app, delete any app folders (e.g. `AppData/Reddit Wallpaper Changer`), then reinstall

Let me know if that helps

**Comment by rkantos** [comment]
I think I have it figured out... A bit of a strange one I guess. First I tried running RWC in Windows Sandbox, which obviously worked fine. After this I figured to have a look with Procmon and noticed the `C:\Users\%USERPROFILE%\AppData\Local\_u_qwertydog\Reddit_Wallpaper_Changer._Url_fn5rdr3tbrupllitik0y5clty5pa2qfl\1.0.20.0` path being read.. In the path there was only a single file:

`user.config` with:
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <userSettings>
        <Reddit_Wallpaper_Changer.Properties.Settings>
            <setting name="AppDataPath" serializeAs="String">
                <value>C:\Users\<username>\AppData\Roaming\Reddit Wallpaper Changer</value>
            </setting>
        </Reddit_Wallpaper_Changer.Properties.Settings>
    </userSettings>
</configuration>
```

I figured why to delete the folder `1.0.20.0`, however the problem persisted. The same folder contains folders for older RWC installs from 1.0.15.0 through 1.0.19.0. Only after renaming the whole folder `Reddit_Wallpaper_Changer._Url_fn5rdr3tbrupllitik0y5clty5pa2qfl` did RWC start normally... 

Finally I dug a little deeper and figured out that renaming the previous version's folder `1.0.19.0` I could make RWC start normally. This is where I found the issue... The previous version's user.config had line full of `NUL` (5400 of them to be exact...) It is still a bit weird since Procmon doesn't appear to show that RWC is actually even trying to read this file.. Only the `user.config` of v. 1.0.20.0 is being read. Deleting/renaming `1.0.19.0` apparently made RWC to restore settings from 1.0.18.0 regardless. Go figure!
